ledger/participant-integration-api/src/main/scala/platform/store/FlywayMigrationsBase.scala

```diff
@@ -0,0 +1,294 @@
+        s"Failed migrations removed: ${migrationsToString(actionsTaken.migrationsRemoved)}. " +
+        s"Missing migrations deleted: ${migrationsToString(actionsTaken.migrationsDeleted)}. " +
+        s"Migration checksums aligned: ${migrationsToString(actionsTaken.migrationsAligned)}."
+    )
+    ()
+  }
+
+  // There is currently no way to get the previous behavior in
+  // a non-deprecated way. See https://github.com/flyway/flyway/issues/3338
+  @nowarn("msg=method ignoreFutureMigrations .* is deprecated")
+  def currentVersion(): Future[Option[String]] = run { configBase =>
+    val flyway = configBase
+      .ignoreFutureMigrations(false)
+      .load()
+    Option(flyway.info.current()).map(_.getVersion.getVersion)
+  }
+
+  // There is currently no way to get the previous behavior in
+  // a non-deprecated way. See https://github.com/flyway/flyway/issues/3338
+  @nowarn("msg=method ignoreFutureMigrations .* is deprecated")
+  def validate(): Future[Unit] = run { configBase =>
+    val flyway = configBase
+      .ignoreFutureMigrations(false)
+      .load()
+    logger.info("Running Flyway validation...")
+    getStatus(flyway) match {
+      case FlywayStatus.SchemaBeforeBaseline(current, baseline) =>
+        throw SchemaOutdated(current, baseline)
+      case FlywayStatus.NeedsBaselineReset(oldVersions) => throw NeedsBaselineReset(oldVersions)
+      case FlywayStatus.DatabaseEmpty(_) => flyway.validate()
+      case FlywayStatus.NoMigrationsPending(_, _) => flyway.validate()
+      case FlywayStatus.Outdated(_, _, _) => flyway.validate()
+    }
+
+    logger.info("Flyway schema validation finished successfully.")
+  }
+
+  @nowarn("msg=method ignoreFutureMigrations .* is deprecated")
+  def migrate(allowExistingSchema: Boolean = false): Future[Unit] = run { configBase =>
+    val flyway = configBase
+      .baselineOnMigrate(allowExistingSchema)
+      .baselineVersion(MigrationVersion.fromVersion("0"))
+      .ignoreFutureMigrations(false)
+      .load()
+    logger.info("Running Flyway migration...")
+    val migrationResult = getStatus(flyway) match {
+      case FlywayStatus.SchemaBeforeBaseline(current, baseline) =>
+        throw SchemaOutdated(current, baseline)
+      case FlywayStatus.NeedsBaselineReset(_) =>
+        resetBaseline(flyway)
+        flyway.migrate()
+      case FlywayStatus.DatabaseEmpty(_) => flyway.migrate()
+      case FlywayStatus.NoMigrationsPending(_, _) => flyway.migrate()
+      case FlywayStatus.Outdated(_, _, _) => flyway.migrate()
+    }
+
+    logger.info(
+      s"Flyway schema migration finished successfully, applying ${migrationResult.migrationsExecuted} steps."
+    )
+  }
+
+  @nowarn("msg=method ignoreFutureMigrations .* is deprecated")
+  def validateAndWaitOnly(retries: Int, retryBackoff: FiniteDuration): Future[Unit] = runF {
+    configBase =>
+      val flyway = configBase
+        .ignoreFutureMigrations(false)
+        .load()
+
+      logger.info("Running Flyway validation...")
```

Comment:
```suggestion
      logger.info("Running Flyway validation with a wait...")
```
So that we can tell them apart in the logfile